openstef/tasks/utils/taskcontext.py

```diff
@@ -16,12 +14,14 @@
 class TaskContext:
     def __init__(
         self,
-        name,
-        suppress_exceptions=False,
-        post_teams_on_exception=True,
-        on_exception=None,
-        on_successful=None,
-        on_end=None,
+        name: str,
+        config,
```

Comment:
Do we always need the config and database for tasks? Maybe we can make it optional if we don't need it for some tasks?